src/lib/checkov/checkov-collector.ts

```diff
@@ -18,16 +18,24 @@ export class CheckovCollector extends AnalysisCollectorBase {
     // Once we develop custom checks, they should be specified using the --external-checks-git argument.
     const args = ['--directory', '.', '--output', 'json', '--no-guide', '--soft-fail'];
     let output;
+    let results = [];
+
     try {
       output = await this.spawn('checkov', args, options);
     }
     catch (e: unknown) {
       throw new Error(`Error executing Checkov: ${e as string}`);
     }
 
-    this.logger.debug(JSON.stringify(output, null, 2));
+    try {
+      this.logger.debug(JSON.stringify(output, null, 2));
+      results = this.parseResults(output);
```

Comment:
```suggestion
      return this.parseResults(output);
```

You _should_ be able to do this. (please check! 😆 )

If you do, then you can remove the `let results = [];` and `return results`.

Generally we prefer to reduce the unnecessary `let xxxx;` variable declarations. If the function gets too big, we can unintentionally assign to these variables.

Reply:
Ok will do this